### PR TITLE
[release-4.9] Bug 2015306: bump api

### PR DESCRIPTION
**Pull request by EmilyM1**
This is to bump the api for backporting atomic-type-bug:
https://github.com/openshift/api/pull/1036
For the changes made in bug 1992592 cluster-config needed an api bump that was done here for 4.10 which is this PR-
https://github.com/openshift/cluster-config-operator/pull/219.
This PR is the corresponding bump for 4.9

1.  If the Bugzilla associated with the PR has the "FastFix" keyword, the subjective assessment on the issue has already been done and a customer is impacted. These PRs should be prioritized for merge.
    - [ ] verified
    - [X] does not apply
2. The bug has significant impact either through severity, reduction in supportability, or number of users affected.
    - [X] verified
    - [ ] does not apply
3. For branches that are in the Maintenance lifecycle phase:
    - [X] The bug is a critical fix, no reasonable workaround exists, and a recommendation for upgrade has been ruled out, or 
    - [ ] The bug is a security related bug
    - [ ] Branch not in maintenance mode yet (current release + previous release for 90 days after current GA; everything older is in maintenance)
4. The severity field of the bug must be set to accurately reflect criticality.
    - [X] verified
5. The PR was created with the cherry-pick bot OR the PR’s description is well formed with user-focused release notes that state the bug number, impact, cause, and resolution. Where appropriate, it should also contain information about how a user can identify whether a particular cluster is affected.
    - [X] verified

### CODE REVIEW

**Comment by stlaz** [comment]
/hold
the bump attempts to bring master@o/api to release 4.8

**Comment by EmilyM1** [comment]
closing while I get the status on 4.9

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc
 	github.com/ghodss/yaml v1.0.1-0.20190212211648-25d852aebe32 // indirect
 	github.com/go-bindata/go-bindata v3.1.2+incompatible
-	github.com/openshift/api v0.0.0-20210817132244-67c28690af52
+	github.com/openshift/api v0.0.0-20211014164657-4436dc8be01e
```
Comment (sttts) [inline go.mod]: if I am not mistaken, 4436dc8be01e is not the release-4.9 branch.

**Comment by sttts** [comment]
Can you squash? The first commit looks wrong, but the later fixes it.

File: vendor/github.com/openshift/api/go.mod [hunk 2]
```diff
@@ -9,9 +9,9 @@ require (
 	github.com/openshift/build-machinery-go v0.0.0-20210712174854-1bb7fd1518d3
 	github.com/spf13/pflag v1.0.5
 	golang.org/x/tools v0.1.2
-	k8s.io/api v0.22.0-rc.0
-	k8s.io/apimachinery v0.22.0-rc.0
-	k8s.io/code-generator v0.22.0-rc.0
+	k8s.io/api v0.22.1
+	k8s.io/apimachinery v0.22.1
+	k8s.io/code-generator v0.22.1
```
Comment (s-urbaniak) [inline vendor/github.com/openshift/api/go.mod:14]: why is the bump to 0.22.1 necessary?

**Comment by s-urbaniak** [comment]
@EmilyM1 please reference the underlying bug in the PR description and add a title.

**Comment by EmilyM1** [comment]
@sur check the new description and the https://github.com/openshift/cluster-config-operator/pull/219 for why this is needed on the backport

File: go.mod [hunk 3]
```diff
@@ -20,8 +20,8 @@ require (
 	google.golang.org/grpc v1.39.1 // indirect
 	gopkg.in/gcfg.v1 v1.2.0
 	gopkg.in/warnings.v0 v0.1.1 // indirect
-	k8s.io/api v0.22.0
-	k8s.io/apimachinery v0.22.0
+	k8s.io/api v0.22.1
+	k8s.io/apimachinery v0.22.1
```
Comment (sttts) [inline go.mod:24]: can we bump all of the kube libs to the same version?

Comment (EmilyM1) [reply go.mod:24]: Sure, I'll try v0.22.1

**Comment by s-urbaniak** [comment]
please squash into one commit, else lgtm

File: vendor/github.com/openshift/api/config/v1/types_feature.go [hunk 4]
```diff
@@ -111,13 +111,14 @@ var FeatureSets = map[FeatureSet]*FeatureGateEnabledDisabled{
...
-		with("CSIMigrationAzureDisk"). // sig-storage, fbertina, Kubernetes feature gate
-		with("ExternalCloudProvider"). // sig-cloud-provider, jspeed, OCP specific
+		with("CSIDriverAzureDisk").         // sig-storage, jsafrane, OCP specific
+		with("CSIDriverVSphere").           // sig-storage, jsafrane, OCP specific
+		with("CSIMigrationAWS").            // sig-storage, jsafrane, Kubernetes feature gate
+		with("CSIMigrationOpenStack").      // sig-storage, jsafrane, Kubernetes feature gate
+		with("CSIMigrationGCE").            // sig-storage, fbertina, Kubernetes feature gate
+		with("CSIMigrationAzureDisk").      // sig-storage, fbertina, Kubernetes feature gate
+		with("ExternalCloudProvider").      // sig-cloud-provider, jspeed, OCP specific
+		with("InsightsOperatorPullingSCA"). // insights-operator/ccx, tremes, OCP specific
```
Comment (s-urbaniak) [inline vendor/github.com/openshift/api/config/v1/types_feature.go:121]: I am not sure this has consequences here, @sttts do you happen to know?

**Review by soltysh — APPROVED** [review]
/approve
/hold cancel

**Pull request merged**